src/daemon/client.rs

```diff
@@ -134,17 +142,46 @@ pub fn run_command(
     socket: &DaemonSocketPath,
     check_alive: bool,
 ) -> Result<()> {
-    if check_alive && !daemon_is_alive(&socket)? {
-        match command {
-            DaemonCommands::Stop => {
-                // in rare cases, daemon_is_alive return false, but the process may still be alive
-                if !ping(&socket)? {
-                    info!("Daemon is not running");
-                    return Ok(());
+    if check_alive {
+        match daemon_is_alive(&socket)? {
+            DaemonStatus::NotRunning => {
+                match command {
+                    DaemonCommands::Stop => {
+                        // in rare cases, daemon_is_alive return false, but the process may still be alive
+                        if !ping(&socket)? {
+                            info!("Daemon is not running");
+                            return Ok(());
+                        }
+                    }
+                    _ => {
+                        return Err(anyhow!("Daemon is not running!"));
+                    }
```

Comment:
check_alive will be false when run_command is called from cli/format.rs etc.
So we can't check match of build id.
Correct implementation is following:
- pass cached DaemonStatus as Option to run_command
- if DaemonStatus is not None, just use this and check build id
- if DaemonStatus is None, re-run daemon_is_alive
  - check_alive arg is no longer necessary